index.rst

```diff
@@ -1,61 +1,461 @@
+
+  p1 = "foo bar=1.0,baz=1.0"
+  p2 = "foo baz=2.0"
+  p3 = "foo bar=3.0,baz=3.0"
+  requests.post(url="http://localhost:8086/write?db=mydb", data=p1)
+  requests.post(url="http://localhost:8086/write?db=mydb", data=p2)
+  requests.post(url="http://localhost:8086/write?db=mydb", data=p3)
+
+The following query returns the ``foo`` measurement:
+
+.. jupyter-execute::
+
+  r = requests.get(url="http://localhost:8086/query", params={'q': 'SELECT * FROM "mydb"."autogen"."foo"'})
+  r.json()['results'][0]['series']
+
+Notice that when querying the foo measurement, InfluxDB returns ``None`` for the missing value of bar in the second point.
+The Python keyword ``None`` is used here as the default value for an optional parameter (the InfluxDB field in this case), as expected.
+
+The InfluxDB Sink connector is the right place for dropping missing values. In particular, the ``influxdb-java`` library used by the connector, `skips fields with NaN values`_ when writing to InfluxDB.
+This feature was implemented upstream a month after we first notice this problem back in September 2019, during the AuxTel integration activities at the Summit.
+
+.. _skips fields with NaN values: https://github.com/influxdata/influxdb-java/blob/master/CHANGELOG.md#216-2019-10-25
+.. _writing NaN values to InfluxDB: https://jira.lsstcorp.org/browse/DM-21300
+
+
+In summary, fields are optional, and we should drop missing values when writing to InfluxDB.
+
+Filling in time intervals with no data
+--------------------------------------
+
+When querying InfluxDB, you can `group the result by time intervals and use fill()`_  to specify how InfluxDB handles time intervals with no data.
+
+.. _group the result by time intervals and use fill(): https://docs.influxdata.com/influxdb/v1.8/query_language/explore-data/#group-by-time-intervals-and-fill
+
+For example, this query will resample the values on a regular time grid of ``10ms`` and use ``fill(linear)`` to perform a linear interpolation for time intervals with no data:
+
+.. jupyter-execute::
+
+  r = requests.get(url="http://localhost:8086/query", params={'q': 'SELECT mean(baz), mean(baz) FROM "mydb"."autogen"."foo" GROUP BY time(10ms) fill(linear)'})
+  r.json()['results'][0]['series']
+
+Writing arrays to InfluxDB
+--------------------------
+
+InfluxDB does not support derived types like arrays, the solution we found was to `extract the array items`_ in the InfluxDB Sink connector and write them to individual fields in InfluxDB.
+
+.. _extract the array items: https://kafka-connect-manager.lsst.io/userguide.html#recording-arrays-in-influxdb
+
+For sparse arrays with ``NaN`` values the connector will extract them and drop the corresponding fields with ``NaN`` values before writing to InfluxDB.
+
+For an Avro record with an array ``[1.0, NaN, 2.0]``, for example,  the connector will first extract the array items into the field set ``foo1=1.0,foo2=NaN,foo3=2.0`` and then write ``foo1=1.0,foo3=3.0`` to InfluxDB.
```

Comment:
I didn't realize this could happen.  I don't think the efd client will handle this gracefully now if some elements of an array are missing.

Reply:
Notice you still can query `foo2` and the missing value should get converted back to `NaN` by the EFD client. I'll try to show that in the following section.